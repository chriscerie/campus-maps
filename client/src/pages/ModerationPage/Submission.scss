.acceptance-buttons-container {
  text-align: right;
}

.acceptance-buttons {
  padding-left: 50px;
}

.submission-link {
  color: #2592ff;
  text-decoration: none;
}

.submission-link:hover {
  color: #2537ff;
}

.submissions-container {
  margin-bottom: 1.5em;
}

.submission-divider {
  width: 90%;
}

.submission-header-container {
  display: table;
  border-spacing: 1.5em 0;
  vertical-align: middle;
  margin: 0 -1.5em 1.5em;
  margin-top: 1.5em;
}

.submission-header-body {
  display: table;
  border-collapse: separate;
  border-spacing: 1.5em 0;
  margin: 0 -1.5em;
}

.submission-profile-image-container {
  display: table-cell;
  border-radius: 50%;
  overflow: hidden;
}

.submission-profile-image {
  height: 4em;
  width: 4em;
  aspect-ratio: auto 1;
  vertical-align: middle;
  box-sizing: border-box;
}

.submission-location-info-container {
  display: table-cell;
  vertical-align: middle;
  width: 100%;
}

.submission-location-info {
  display: block;
  font-size: larger;
  font-weight: 700;
}

.submission-text-container {
  width: 90%;
}

.submission-text {
  margin: 0;
  font-size: smaller;
  line-height: 1.7em;
}

#submission-accept-button {
  color: #fff;
  background-color: #2592ff;
  margin-left: 5px;
  margin-right: 5px;
  border: 1px solid #2592ff;
  padding: 0.4em 1.4em;
  border-radius: 0.3em;
  font-weight: bold;
  font-size: 1em;
  text-transform: none;
}

#submission-accept-button:hover {
  background-color: #2537ff;
  border: 1px solid #2537ff;
}

#submission-reject-button {
  color: #929292;
  background-color: #fff;
  margin-left: 5px;
  margin-right: 5px;
  border: 1px solid #929292;
  padding: 0.4em 1.4em;
  border-radius: 0.3em;
  font-weight: bold;
  font-size: 1em;
  text-transform: none;
}

#submission-reject-button:hover {
  background-color: #929292;
  color: #fff;
}
