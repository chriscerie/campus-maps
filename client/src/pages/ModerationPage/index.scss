.align-text-left {
  text-align: left;
}

.header-text {
  font-size: 3em;
  font-weight: 700;
  line-height: 1.2em;
  letter-spacing: -0.4px;
  margin-top: 40px;
  margin-bottom: 32px;
}
.submissions {
  display: block;
  padding-top: 10px;
  padding-bottom: 32px;
}

.moderation-page-container {
  display: block;
  line-height: 2em;
  min-height: 100vh;
}

#profile-greeting-icon {
  float: left;
  width: 32px;
  height: 32px;
}

.profile-greeting-text {
  font-size: 1em;
  margin-left: 1em;
}
