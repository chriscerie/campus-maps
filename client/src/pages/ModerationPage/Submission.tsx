import { Button, Divider } from '@mui/material';
import type { UserType } from '../../types/UserType';
import { Fragment } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import './Submission.scss';

export type SubmissionType = {
  _id: string;
  author_id: string;
  id: string;
  name: string;
  type: string;
  description: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  zip_code: string;
  rooms: Array<{
    room_name: string;
    room_id: string;
  }>;
  __v: string;
  author?: UserType;
};

const Submission = ({
  submission,
  resetSubmissions,
}: {
  submission: SubmissionType;
  resetSubmissions: () => void;
}) => {
  return (
    <Fragment>
      <li className="submissions-container">
        <Divider className="submission-divider" />

        <div className="submission-header-container">
          <div>
            <div className="submission-header-body">
              <div className="submission-profile-image-container">
                <img
                  src={
                    submission.author ? submission.author.profile_picture : ''
                  }
                  alt="profile"
                  className="submission-profile-image"
                />
              </div>
              <div className="submission-location-info-container">
                <div className="submission-location-info">
                  <Link
                    className="submission-link"
                    to={`/loc/${submission.id}`}
                  >
                    {submission.name}
                  </Link>
                </div>
                <div>
                  Update by {submission.author?.first_name}{' '}
                  {submission.author?.last_name}
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="submission-text-container">
          <p className="submission-text">
            <strong>Type:</strong> {submission.type} <br />
            <strong>Description:</strong> {submission.description} <br />
            <strong>Address 1:</strong> {submission.address1} <br />
            <strong>Address 2:</strong> {submission.address2} <br />
            <strong>City:</strong> {submission.city} <br />
            <strong>State:</strong> {submission.state} <br />
            <strong>ZIP Code:</strong> {submission.zip_code} <br />
            <strong>Rooms:</strong> <br />
            {submission.rooms &&
              submission.rooms.map((room) => (
                <div>
                  {'â€¢ ' + room.room_name}
                  <br />
                </div>
              ))}
          </p>
          <div className="acceptance-buttons-container">
            {/* Buttons are nonfunctional currently */}
            <Fragment>
              <Button
                variant="contained"
                disableElevation
                id="submission-accept-button"
                onClick={() => {
                  axios
                    .post(`/api/v1/locations/moderation/${submission._id}`, {
                      decision: 'Accept',
                    })
                    .then(() => {
                      resetSubmissions();
                    });
                }}
              >
                Accept
              </Button>
              <Button
                variant="contained"
                disableElevation
                id="submission-reject-button"
                onClick={() => {
                  axios
                    .post(`/api/v1/locations/moderation/${submission._id}`, {
                      decision: 'Deny',
                    })
                    .then(() => {
                      resetSubmissions();
                    });
                }}
              >
                Reject
              </Button>
            </Fragment>
          </div>
        </div>
      </li>
    </Fragment>
  );
};

export default Submission;
