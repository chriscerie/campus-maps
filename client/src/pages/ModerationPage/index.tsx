import { Avatar } from '@mui/material';
import { Grid, Container } from '@mui/material';
import React, { Fragment, useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '../../reducers';
import axios from 'axios';
import type { UserType } from '../../types/UserType';
import { SubmissionType } from './Submission';
import Submission from './Submission';
import './index.scss';

const setSubmissions = (
  setBackendSubmissions: React.Dispatch<React.SetStateAction<SubmissionType[]>>
) => {
  axios
    .get<Array<SubmissionType>>('/api/v1/locations/loc-edit')
    .then(async (res) => {
      setBackendSubmissions(
        await Promise.all(
          res.data.map(async (submission) => {
            const res = await axios.get<UserType>(
              `/api/v1/users/${submission.author_id}`
            );
            submission.author = res.data;
            return submission;
          })
        )
      );
    });
};

function ModerationPage() {
  const currentUser = useSelector((state: RootState) => state.currentUser);
  const [backendSubmissions, setBackendSubmissions] = useState<
    Array<SubmissionType>
  >([]);

  useEffect(() => {
    axios
      .get<Array<SubmissionType>>('/api/v1/locations/loc-edit')
      .then(async (res) => {
        setSubmissions(setBackendSubmissions);
      });
  }, []);

  return (
    <div className="moderation-page-container align-text-left">
      {currentUser && currentUser.account_type === 'Admin' ? (
        <Fragment>
          <Container>
            <div className="header-text">Moderation Panel</div>
            <Grid container>
              <Grid item xs={12}>
                <div className="profile-icon-container">
                  <Avatar
                    id="profile-greeting-icon"
                    src={currentUser ? currentUser.profile_picture : ''}
                  />
                  <div className="profile-greeting-text">
                    You are logged in as {currentUser.first_name}{' '}
                    {currentUser.last_name}.
                  </div>
                </div>
              </Grid>
              <Grid item xs={12}>
                <div className="submissions">
                  <h1>Pending Submissions</h1>
                  <ul>
                    {backendSubmissions.map((rootComment: SubmissionType) => (
                      <Submission
                        key={rootComment._id}
                        submission={rootComment}
                        resetSubmissions={() => {
                          setSubmissions(setBackendSubmissions);
                        }}
                      />
                    ))}
                  </ul>
                </div>
              </Grid>
            </Grid>
          </Container>
        </Fragment>
      ) : (
        <Fragment>
          <Container>
            <div className="header-text">Moderation</div>
            <Grid container>
              <Grid item xs={12}>
                <div className="profile-icon-container">
                  <Avatar id="profile-greeting-icon" src={''} />
                  <div className="profile-greeting-text">
                    You do not have permission to view this page.
                  </div>
                </div>
              </Grid>
            </Grid>
          </Container>
        </Fragment>
      )}
    </div>
  );
}

export default ModerationPage;
