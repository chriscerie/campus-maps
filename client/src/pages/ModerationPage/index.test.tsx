import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers';
import ModerationPage from './index';

describe('ModerationPage', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ModerationPage />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
