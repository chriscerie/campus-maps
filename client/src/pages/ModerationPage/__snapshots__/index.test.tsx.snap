// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModerationPage should render successfully 1`] = `
<div
  className="moderation-page-container align-text-left"
>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1oqqzyl-MuiContainer-root{width:100%;margin-left:auto;box-sizing:border-box;margin-right:auto;display:block;padding-left:16px;padding-right:16px;}@media (min-width:600px){.css-1oqqzyl-MuiContainer-root{padding-left:24px;padding-right:24px;}}@media (min-width:1200px){.css-1oqqzyl-MuiContainer-root{max-width:1200px;}}",
      }
    }
    data-emotion="css 1oqqzyl-MuiContainer-root"
  />
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="header-text"
    >
      Moderation
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-11lq3yg-MuiGrid-root{box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex-wrap:wrap;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;width:100%;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}",
        }
      }
      data-emotion="css 11lq3yg-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1em9ut6-MuiGrid-root{box-sizing:border-box;margin:0;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:100%;}",
          }
        }
        data-emotion="css 1em9ut6-MuiGrid-root"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
      >
        <div
          className="profile-icon-container"
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-2s90m6-MuiAvatar-root{position:relative;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;width:40px;height:40px;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-size:1.25rem;line-height:1;border-radius:50%;overflow:hidden;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;color:#fff;background-color:#bdbdbd;}",
              }
            }
            data-emotion="css 2s90m6-MuiAvatar-root"
          />
          <div
            className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            id="profile-greeting-icon"
          >
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-1i6ghfn-MuiAvatar-fallback{width:75%;height:75%;}",
                }
              }
              data-emotion="css 1i6ghfn-MuiAvatar-fallback"
            />
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;width:75%;height:75%;}",
                }
              }
              data-emotion="css 10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            className="profile-greeting-text"
          >
            You do not have permission to view this page.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
